### ISSUE TITLE
Support for arbitary depth heading

### ISSUE THREAD

**Issue by wrvsrx** [treeman/tree-sitter-djot#53]
Currently heading deeper than 6 is not supoprted

```djot
######### j
```

**Comment by treeman** [comment]
Hmm. Do you have a specific use-case for depth 7+? The markdown treesitter also only supports 6 deep.

It's possible to support arbitrary nesting, the problem is that we create tokens like `heading1`, `heading2` etc and as far as I know they can't be created dynamically with treesitter. So either we abandon them and only use `heading`, relying on Neovim's `#eq?` predicate for queries (I hope there are similar ones for other editors), or we add a `heading_deep` that captures headings of 7+.

I don't know what approach I think is the best.

**Comment by wrvsrx** [comment]
I mention that just because djot support that in specification.

For tree-sitter highlighting, I think `heading` is enough?

**Comment by treeman** [comment]
You're right, this is something we should support. I'm leaning towards `heading` being the best approach...
